packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -129,23 +176,33 @@ export const Picker: React.FC<IPickerProps> = ({
           isError={!!error}
           isOpen={isListOpen}
           isDisabled={disabled}
-          isItemSelected={!!selectedItem}
+          isItemSelected={!!selected}
           isRequired={isRequired}
+          isMultiSelect={type === 'multi'}
           size={size}
-          onClick={handleOnTriggerClick}
-          onClearClick={handleOnClearClick}
+          onTrigger={handleOnTrigger}
+          onClear={handleOnClear}
           onFilter={handleOnFilter}
         >
-          {selectedItem ? selectedItem.name : placeholder}
+          {selected ? (
```

Comment:
It will trigger also for empty array, is this correct here?

Reply:
Good catch. After we removed the initial state of selected items, we could expect that an empty array can be passed. I will add an additional check here.